e2e/rgraph_update_action_test.go

```diff
@@ -0,0 +1,262 @@
+		x.TimeoutSec = 6
+		x.Type = "HTTP"
+		x.HttpHealthCheck = &compute.HTTPHealthCheck{
+			RequestPath: "/",
+			Port:        int64(9376),
+		}
+	})
+	hcRes, err := hcMutRes.Freeze()
+	if err != nil {
+		return nil, err
+	}
+	hcBuilder := healthcheck.NewBuilder(hcID)
+	hcBuilder.SetOwnership(rnode.OwnershipManaged)
+	hcBuilder.SetState(rnode.NodeExists)
+	hcBuilder.SetResource(hcRes)
+	graphBuilder.Add(hcBuilder)
+	return hcID, nil
+}
+
+func TestHcUpdateWithBackendService(t *testing.T) {
+	ctx := context.Background()
+	graphBuilder := rgraph.NewBuilder()
+
+	hcID, err := buildHealthCheck(graphBuilder, "hc-test", 15)
+	if err != nil {
+		t.Fatalf("buildHealthCheck(_, hc-test, _) = %v, want nil", err)
+	}
+
+	bsID, err := buildBackendService(graphBuilder, "bs-e2e", hcID, 80)
+	if err != nil {
+		t.Fatalf("buildBackendService(_, bs-e2e, _, 80) = %v, want nil", err)
+	}
+
+	graph, err := graphBuilder.Build()
+	if err != nil {
+		t.Fatalf("graphBuilder.Build() = %v, want nil", err)
+	}
+	result, err := plan.Do(ctx, theCloud, graph)
+	if err != nil {
+		t.Fatalf("plan.Do(_, _, _) = %v, want nil", err)
+	}
+
+	var viz exec.GraphvizTracer
+	ex, err := exec.NewSerialExecutor(result.Actions, exec.TracerOption(&viz))
+	if err != nil {
+		t.Logf("exec.NewSerialExecutor(_, _) err: %v", err)
+		return
+	}
+	res, err := ex.Run(context.Background(), theCloud)
+	if err != nil || res == nil {
+		t.Errorf("ex.Run(_,_) = %v, want nil", err)
+	}
+
+	t.Cleanup(func() {
+		err = theCloud.BackendServices().Delete(ctx, bsID.Key)
+		if err != nil {
+			t.Logf("delete backend service: %v", err)
+		}
+		err := theCloud.HealthChecks().Delete(ctx, hcID.Key)
+		if err != nil {
+			t.Logf("delete health check: %v", err)
+		}
+	})
+	checkGCEResources(t, ctx, hcID, bsID, 15)
+
+	// update health check
+	hcID, err = buildHealthCheck(graphBuilder, "hc-test", 60)
+	if err != nil {
+		t.Fatalf("buildHealthCheck(_, hc-test, 60) = (_, %v), want (_, nil)", err)
+	}
+	graph, err = graphBuilder.Build()
+	if err != nil {
+		t.Fatalf("graphBuilder.Build() = %v, want nil", err)
+	}
+	result, err = plan.Do(ctx, theCloud, graph)
+	if err != nil {
+		t.Fatalf("plan.Do(_, _, _) = %v, want nil", err)
+	}
+	expectedActions := []exec.ActionType{
+		exec.ActionTypeUpdate,
+	}
+
+	t.Logf("\nPlan.Actions: %v", result.Actions)
+	t.Logf("\nPlan.Got: %v", result.Got)
+	t.Logf("\nPlan.Want: %v", result.Want)
+
+	err = expectActions(result.Actions, expectedActions)
+	if err != nil {
+		t.Fatalf("expectActions(_, _) = %v, want nil", err)
+	}
+	t.Log("\nstart NewSerialExecutor for update")
+	ex, err = exec.NewSerialExecutor(result.Actions, exec.TracerOption(&viz))
+	if err != nil {
+		t.Logf("exec.NewSerialExecutor err: %v", err)
+		return
+	}
+	res, err = ex.Run(context.Background(), theCloud)
+	if err != nil || res == nil {
+		t.Errorf("ex.Run(_,_) = ( %v, %v), want (*result, nil)", res, err)
+	}
+	t.Logf("exec.NewSerialExecutor finished, res: %v", res)
+	if len(res.Pending) > 0 {
```

Comment:
nit: as discussed it shouldn't ever occur with this executor. Otherwise, you may consider adding similar check for the previous `ex.Run` call